app/src/marketplace/components/StreamPreviewPage/StreamLivePreview/index.jsx

```diff
@@ -218,16 +219,17 @@ export class StreamLivePreview extends Component<Props, State> {
                                                     </tr>
                                                 </thead>
                                                 <tbody>
-                                                    {data.map((d) => (
+                                                    {data.map((d, index) => (
                                                         <tr
-                                                            key={d.metadata.offset}
+                                                            // eslint-disable-next-line react/no-array-index-key
+                                                            key={`${d.metadata.messageId.timestamp}-${index}`}
```

Comment:
This should probably be: `d.metadata.messageId.serialize()`

Reply:
Alternatively, just use `index`? I don't know what this `key` really does since it doesn't maintain identity when items get reordered, nor does it maintain the row's identity i.e. same row but different data, and it says weird stuff about identity in the case that there's sequential items with the same timestamp

e.g. if the timestamp and index are [timestamp, index]

State A:

`message3`: [0, 0] 
`message2`: [0, 1]
`message1`: [0, 2]

State B:

`message4`: [1, 0] 
`message3`: [0, 1] 
`message2`: [0, 2]
`message1`: [0, 3]

Then you'll see that the key for `message2` and `message3` is the same between renders.

Reply:
I've gone with stringifying the messageId here @timoxley - AFAIK React needs more render cycles to produce a list with dynamic inserts using the array index which is why I'm looking for a source of uniqueness here.